package bigit;
//option java_package = "com.bigit.message";
message DeviceInfo{
	required string name = 1;
	required string sn = 2;
	required string cpu = 3;
	required string brand = 4;
	required string model = 5;
	required string imei = 6;
	optional string mac = 7;
}

message StorageInfo{
	required string total = 1;
	required string free = 2;
	required string used = 3;
	
}

message AppInfo{
	required string id = 1;
	required string name = 2;
	required string version = 3;
	required string size = 4;
	required string location = 5;
	required string icodata = 6;
}

message AppList{
	repeated AppInfo app = 1;
}

message ResInfo{
	required string path = 1;
	required string size = 2;
	required string icodata = 3;
	optional string createdate = 4;
	optional string duration = 5;
	optional string author = 6;
	optional string format = 7;
}

message ResList{
	repeated ResInfo res = 1;
}

message Person {
  required string name = 1;
  required string id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

message AddressBook {
  repeated Person person = 1;
 
}

message SMSInfo{
  enum SMSType {
    IN = 0;
    OUT = 1;
  }
  
	required string content = 1;
	repeated string numbers = 2;
	required SMSType type = 3;
}

message SMSList {
  repeated SMSInfo sms = 1;
 
}

message test
{
 required string msg = 1;
}
